src/resolvers/shipment/index.ts

```diff
@@ -0,0 +1,68 @@
+import { UserInputError } from 'apollo-server'
+
+import {
+  QueryResolvers,
+  MutationResolvers,
+  ShipmentResolvers,
+  ShipmentInput,
+} from '../../server-internal-types'
+import { sequelize } from '../../sequelize'
+import Shipment from '../../models/shipment'
+import Group from '../../models/group'
+
+const shipmentRepository = sequelize.getRepository(Shipment)
+const groupRepository = sequelize.getRepository(Group)
+
+// Shipment query resolvers
+const listShipments: QueryResolvers['listShipments'] = async () => {
+  return shipmentRepository.findAll()
+}
+
+// Shipment mutation resolvers
+const addShipment: MutationResolvers['addShipment'] = async (
+  _parent,
+  { input },
+  _context,
+) => {
+  if (!input) {
+    throw new UserInputError('Shipment arguments invalid', {
+      invalidArgs: 'input',
+    })
+  }
+
+  if (
+    !input.shippingRoute ||
+    !input.labelYear ||
+    !input.labelMonth ||
+    !input.sendingHubId ||
+    !input.receivingHubId ||
+    !input.status
+  ) {
+    throw new UserInputError('Shipment arguments invalid', {
+      invalidArgs: Object.keys(input).filter(
+        (key) => !input[key as keyof ShipmentInput],
+      ),
+    })
+  }
```

Comment:
I wasn't sure the best way to handle validation. Happy to have others input here.

Reply:
I recommend to use validate using JSON schema, which is language-agnostic and has very powerful validation rules.
[ajv](https://github.com/ajv-validator/ajv) is a good choice for that.
Combining it with [@sinclair/typebox](https://github.com/sinclairzx81/typebox) then turns the JSON schema into TypeScript types.
[Here is an example from a lambda where I use it](https://github.com/NordicSemiconductor/asset-tracker-cloud-aws-js/blob/e686209377b8864356d3147968e5b73a6a332b5b/cellGeolocation/httpApi/addCellGeolocation.ts#L60)

Reply:
In addition, this code should also then validate that the passed ids are valid, e.g. the request should be rejected if the receiving hub does not exist, and/or the user does not have sufficient permissions to create a shipment.

Reply:
Suggestion:  let's leave permissions out of this PR since we haven't set up Auth0 yet (Bion is working on it in #42).

Reply:
@coderbyheart good point about validating the existence of the sending and receiving hub. I can add that to this PR. I agree with @deammer that permission validation can probably wait for @bion's work, but I'll also let @bion weigh in on this as well.

This JSON schema validation seems like a good way to go. I'll be doing a second PR to add two more shipment CRUD resolvers and would be happy to dig into better validation then.

- [ ] TODO: Validate that sending and receiving hub exist before db write

Reply:
@coderbyheart's recommendations of ajv with @sinclair/typebox sound good to me for improving validations.

For permissions yes let's wait until authentication is available before implementing authorization. Hopefully my auth0 PR will be done soon!

Reply:
Ticket created to track API input validation work: https://github.com/distributeaid/shipment-tracker/issues/47